src/components/AmountInput.tsx

```diff
@@ -1,35 +1,65 @@
-import clsx from 'clsx';
+import { useState } from 'react';
+import { useAtomValue } from 'jotai';
+import { displayFunctionsAtom } from 'store/app';
+import { stringifyValue, parseAsValue } from '@agoric/ui-components';
+import { AssetKind } from '@agoric/ertp';
+import type { Brand } from '@agoric/ertp/src/types';
+import StyledInput from './StyledInput';
 
 type Props = {
+  value?: bigint | null;
   label?: string;
   error?: string;
+  brand?: Brand | null;
+  onChange: (value: bigint) => void;
   disabled?: boolean;
 };
 
-const AmountInput = ({ label, error, disabled = false }: Props) => {
+const AmountInput = ({
+  value,
+  onChange,
```

Comment:
looks like this is mixing functionality of controlled input and an uncontrolled one. `value` comes in but `fieldString` and thus `displayString` is local state.

reasonable to clean this up?

Reply:
Added a comment, this one is kind of tricky, PTAL